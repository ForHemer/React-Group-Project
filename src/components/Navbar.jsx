import { NavLink } from 'react-router-dom';
import logo from './assets/planet.png';

const Navbar = () => (

  <header>
    <nav>
      <div className="header-container">
        <img className="logo" src={logo} alt="logo" />
        <h1 className="header-title">Space Travelers&apos; Hub</h1>
      </div>
      <ul>
        <li>
          <NavLink
            to="/"
            className="rockets-page"
          >
            Rockets
          </NavLink>
        </li>
        <li>
          <NavLink
            to="missions"
          >
            Missions
          </NavLink>
        </li>
        <li>
          <NavLink
            to="myprofile"
            className="my-profile-nav"
          >
            My Profile
          </NavLink>
        </li>
      </ul>
    </nav>
  </header>
);

export default Navbar;
