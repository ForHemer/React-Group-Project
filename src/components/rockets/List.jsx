import { useDispatch, useSelector } from 'react-redux';
import { useEffect } from 'react';
import Rocket from './Rocket';
import { getRockets, rocketAPICall } from '../../redux/rockets/rockets';

const RocketList = () => {
  const dispatch = useDispatch();
  const rocketStorage = useSelector((state) => state.rocketReducer);

  useEffect(() => {
    async function getRocketData() {
      if (rocketStorage === null) {
        const fetch = await rocketAPICall();
        dispatch(getRockets(fetch));
      }
    }
    getRocketData();
  }, []);

  return (
    <div>
      {(rocketStorage || []).map((rocket) => (
        <Rocket
          name={rocket.name}
          image={rocket.image}
          key={rocket.id}
          id={rocket.id}
          description={rocket.description}
          reserved={rocket.reserved}
        />
      ))}
    </div>
  );
};

const DisplayedRockets = () => (
  <section>
    <RocketList />
  </section>
);

export default DisplayedRockets;
