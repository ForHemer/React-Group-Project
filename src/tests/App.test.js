import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import List from '../components/rockets/List';
import store from '../redux/configureStore';
import Navbar from '../components/Navbar';
import MissionList from '../components/missions/MissionsList';
import Profile from '../components/profile/MyProfile';

describe('Testing Rendering', () => {
  test('Rockets rendering check', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <List />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Missions rendering check', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <MissionList />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Profile rendering check', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Profile />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Navbar rendering check', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <Navbar />
        </Router>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
