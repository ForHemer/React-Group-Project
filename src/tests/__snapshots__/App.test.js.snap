// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Rendering Navbar rendering check 1`] = `
<header>
  <nav>
    <div
      className="header-container"
    >
      <img
        alt="logo"
        className="logo"
        src="planet.png"
      />
      <h1
        className="header-title"
      >
        Space Travelers' Hub
      </h1>
    </div>
    <ul>
      <li>
        <a
          aria-current="page"
          className="rockets-page active"
          href="/"
          onClick={[Function]}
        >
          Rockets
        </a>
      </li>
      <li>
        <a
          className=""
          href="/missions"
          onClick={[Function]}
        >
          Missions
        </a>
      </li>
      <li>
        <a
          className="my-profile"
          href="/myprofile"
          onClick={[Function]}
        >
          My Profile
        </a>
      </li>
    </ul>
  </nav>
</header>
`;

exports[`Testing Rendering Profile rendering check 1`] = `
<section>
  <div
    className="my-rockets"
  >
    <h2
      className="rocket-title"
    >
      My Rockets
    </h2>
    <ul
      className="rocket-list"
    />
  </div>
</section>
`;

exports[`Testing Rendering Render Navbar 1`] = `
<header>
  <nav>
    <div
      className="header-container"
    >
      <img
        alt="logo"
        className="logo"
        src="planet.png"
      />
      <h1
        className="header-title"
      >
        Space Travelers' Hub
      </h1>
    </div>
    <ul>
      <li>
        <a
          aria-current="page"
          className="rockets-page active"
          href="/"
          onClick={[Function]}
        >
          Rockets
        </a>
      </li>
      <li>
        <a
          className=""
          href="/missions"
          onClick={[Function]}
        >
          Missions
        </a>
      </li>
      <li>
        <a
          className="my-profile"
          href="/myprofile"
          onClick={[Function]}
        >
          My Profile
        </a>
      </li>
    </ul>
  </nav>
</header>
`;

exports[`Testing Rendering Render Profile section 1`] = `
<section>
  <div
    className="my-rockets"
  >
    <h2
      className="rocket-title"
    >
      My Rockets
    </h2>
    <ul
      className="rocket-list"
    />
  </div>
</section>
`;

exports[`Testing Rendering Render Rockets section 1`] = `
<section>
  <div />
</section>
`;

exports[`Testing Rendering Rockets rendering check 1`] = `
<section>
  <div />
</section>
`;
